### PR TITLE
docs: added new advanced filtering text and images

### PR DESCRIPTION
**Pull request by Solly2**
Added new wording and text to the filtering pattern.

This adds the "advanced filtering" section as an extension on the filtering pattern.

### CODE REVIEW

**Comment by vikkipaterson** [comment]
I will also add to this that the integration dev team are building code for this :)

**Comment by Solly2** [comment]
Here is a link to the relevant component in our Storybook - https://pages.github.ibm.com/cloud-integration-design/ide-components/?path=/story/ide-filter-under-construction-%F0%9F%9A%A7--default-example

File: src/pages/patterns/filtering/index.mdx [hunk 1]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+
+_Advanced search_ can be used when the user requires to search and filter a list of items by choosing more than one filter from multiple filter categories. This pattern is especially useful when the number of items within each category is dynamic, such as filtering a table by a user or user defined tags.
+
+Advanced search is also commonly used when there is not the requried amount of space on the screen to accommodate a vertical list.
+
+<br />
+<Row>
+<Column colLg={8}>
+
+![Contextual image](images/filter-8.png)
```
Comment (connor-leech) [inline src/pages/patterns/filtering/index.mdx]: For alt text, please use more detailed descriptions of what's in the image. It doesn't have to be long, but "Contextual image" doesn't provide much utility to a visually impaired reader using a screen reader.

File: src/pages/patterns/filtering/index.mdx [hunk 2]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+<br />
+<br />
+
+The filtering categories appear as headers within a dropdown space underneath a search field, with the top five items from within each section appearing as defaults within this space, along with the expected number of returns if the user was to select that option.
+
+<br />
+<Row>
+<Column colLg={8}>
+
+![Contextual image](images/filter-9.png)
```
Comment (connor-leech) [inline src/pages/patterns/filtering/index.mdx]: Same note about alt text.

File: src/pages/patterns/filtering/index.mdx [hunk 3]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+<br />
+<br />
+
+As the user types into the search field the categories of filters are themselves filtered by the text entered, with the matching text being highlighted.
+
+<br />
+<Row>
+<Column colLg={8}>
+
+![Contextual image](images/filter-10.png)
```
Comment (connor-leech) [inline src/pages/patterns/filtering/index.mdx]: Same note about alt text.

File: src/pages/patterns/filtering/index.mdx [hunk 4]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+<br />
+<br />
+
+Entering a search term which exactly matches an item within a drop down enters that filter into the search field as a tag with the appropriate label attached. Alternatively a user can select that item directly from the drop down by clicking on it. If the term entered into the search field doesn't match any values within the filter categories then that term is entered as a free text search.
+
+<br />
+<Row>
+<Column colLg={8}>
+
+![Contextual image](images/filter-11.png)
```
Comment (connor-leech) [inline src/pages/patterns/filtering/index.mdx]: Same note about alt text.

**Comment by vikkipaterson** [comment]
@connor-leech is there any progress on this PR?

**Comment by connor-leech** [comment]
@vikkipaterson Waiting on Chris' review, but he's out of town until Monday

**Comment by connor-leech** [comment]
@chrisconnors-ibm Any chance you can take a look at this this week?

File: src/pages/patterns/filtering/index.mdx [hunk 5]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
 </Column>
 </Row>
 
+### Advanced filtering
+
+_Advanced search_ can be used when the user requires to search and filter a list of items by choosing more than one filter from multiple filter categories. This pattern is especially useful when the number of items within each category is dynamic, such as filtering a table by a user or user defined tags.
+
+Advanced search is also commonly used when there is not the requried amount of space on the screen to accommodate a vertical list.
```
Comment (chrisconnors-ibm) [inline src/pages/patterns/filtering/index.mdx]: replace:
"Filtered search is also recommended when there is not enough space to accommodate a vertical list of filters."

File: src/pages/patterns/filtering/index.mdx [hunk 6]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+<Column colLg={8}>
+
+![Advanced filter input field example](images/filter-8.png)
+
+</Column>
+</Row>
+<br />
+<br />
+
+The filtering categories appear as headers within a dropdown space underneath a search field, with the top five items from within each section appearing as defaults within this space, along with the expected number of returns if the user was to select that option.
```
Comment (chrisconnors-ibm) [inline src/pages/patterns/filtering/index.mdx]: what's the significance of the top 5?

File: src/pages/patterns/filtering/index.mdx [hunk 7]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+<Column colLg={8}>
+
+![The dropdown with a text filter applied](images/filter-10.png)
+
+</Column>
+</Row>
+<br />
+<br />
+
+Entering a search term which exactly matches an item within a drop down enters that filter into the search field as a tag with the appropriate label attached. Alternatively a user can select that item directly from the drop down by clicking on it. If the term entered into the search field doesn't match any values within the filter categories then that term is entered as a free text search.
```
Comment (chrisconnors-ibm) [inline src/pages/patterns/filtering/index.mdx]: replace with:
Entering a search term which exactly matches an item within a drop down enters that filter into the search field as a tag with the appropriate label attached; a user may also select that item directly from the drop down by clicking on it. If the term entered into the search field doesn't match any values within the filter categories, that term is entered as a free text search, not in a tag.

File: src/pages/patterns/filtering/index.mdx [hunk 8]
```diff
@@ -69,6 +69,62 @@ Multiple category selection is most commonly manifested in either a vertical lis
...
+
+<br />
+<Row>
+<Column colLg={8}>
+
+![Example of selectred search terms having been entered](images/filter-11.png)
```
Comment (chrisconnors-ibm) [inline src/pages/patterns/filtering/index.mdx]: the free text search term should not be in a tag.

File: src/pages/patterns/filtering/index.mdx (hunk 6 above)
Comment (Solly2) [reply src/pages/patterns/filtering/index.mdx]: There is no real significance to the number of items shown, just aware that we need to give an example of the things you can filter by.

If there are a greater number of 'category' of filters then 5 would be quite cumbersome.

Perhaps we can simply change it to "with an example from each section appearing as defaults within this space"

**Comment by vikkipaterson** [comment]
@connor-leech are you waiting on something from our end on this?

**Comment by Solly2** [comment]
I've made the changes as Chris requested, for some reason the build failed though...  All there was were text changes so should be finbe.

There is a bit of an unanswered question around what should populate the drop down though...

**Comment by connor-leech** [comment]
As all existing production patterns must be reworked and revisited, closing this PR and incorporating relevant content into a new doc.

Please continue any discussion here: https://github.com/carbon-design-system/carbon-website/issues/712

**Pull request closed without merging**